### PR TITLE
[PLATFORM-216] Add Dropdown component for filters

### PR DESCRIPTION
**Pull request by tumppi**
Added new `Dropdown` component which handles state changes so it's much easier to use.

Also added needed filters for Canvas listing using this component.

### CODE REVIEW

File: app/src/shared/components/Dropdown/DropdownItem/index.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+type Props = {
+    children: Node,
+    value: string,
+    index?: number,
+    selected?: boolean,
+    onClick?: (index: number) => void,
+}
+
+const TickIcon = (props: any) => (
```
[the remark this answers is not in the document]
Comment (timoxley) [reply app/src/shared/components/Dropdown/DropdownItem/index.jsx]: A story that showed all the available icons would be neat.

[the remark this answers is not in the document]
Comment (mondoreale) [reply app/src/shared/components/Dropdown/DropdownItem/index.jsx]: @tumppi can you try out the new `SvgIcon` I just merged? Let me know if it works, or if doesn't.

[the remark this answers is not in the document]
Comment (tumppi) [reply app/src/shared/components/Dropdown/DropdownItem/index.jsx]: @mondoreale: Will do!

File: app/src/shared/components/Dropdown/DropdownItem/index.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    render() {
+        const { children, selected } = this.props
+
+        return (
+            <RsDropdownItem
+                tag="a"
+                href="#"
+                onClick={this.onClick}
+                className={classNames(styles.dropdownItem, {
+                    active: selected,
```
[the remark this answers is not in the document]
Comment (tumppi) [reply app/src/shared/components/Dropdown/DropdownItem/index.jsx]: Actually we don't need the `selected` style at all for the item. I'll rename the prop to `active` 👍

**Comment by tumppi** [comment]
@juhah: I added story for this component but the styles are a bit different in Storybook vs running with `npm start`. Has something to do with reactstrap. We can fix it later.

**Pull request merged**